hledger/Hledger/Cli/Utils.hs

```diff
@@ -117,6 +118,31 @@ journalApplyValue ropts j = do
                 = id
     return $ convert j
 
+-- | Run PeriodicTransactions from journal from today or journal end to requested end day.
+-- Add generated transactions to the journal
+journalAddForecast :: CliOpts -> Journal -> IO Journal
+journalAddForecast opts j = do
+  today <- getCurrentDay
+  let startDate =
+        -- Create forecast starting from today or end of journal, whatever is earliest, and until the end of requested reporting period
+        -- If end is not provided, do 180 days of forecast
+        case spanEnd (jdatespan j) of
+          Just jEnd -> min jEnd today
+          Nothing -> today
+      endDate = fromMaybe (addDays 180 today) $ periodEnd (period_ ropts)
```

Comment:
Why we need this ~6 extra months outside of report period?

Reply:
I'm of the opinion that for the best result end date have to be explicitly specified, but if not, there should be some sensible default. Not sure that 180 days is sensible, though.

Note that it only applies if -e is not given

Reply:
Oh. Sorry I've missed that it actually adds a default report end date.

> I'm of the opinion that for the best result end date have to be explicitly specified, but if not, there should be some sensible default. Not sure that 180 days is sensible, though.

I'm not sure if it makes sense but I thought that it might be something like `startDate` (forecast) + "longest period among all periodic transactions".
But it will require more changes to get `Interval` parsing extracted out of `runPeriodicTransaction`.

So I guess it should be fine to leave 180 days as a default value for `--forecast` without explicit `-e`.